translate/snippets/detect.go

```diff
@@ -0,0 +1,43 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package trsnip contains snippet code for the Translate API.
+// The code is not runnable.
+package trsnip
+
+// [START translate_detect_language]
+
+import (
+	"context"
+	"fmt"
+
+	"cloud.google.com/go/translate"
+)
+
+func detectLanguage(text string) (*translate.Detection, error) {
+	ctx := context.Background()
+	client, err := translate.NewClient(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("translate.NewClient: %v", err)
+	}
+	defer client.Close()
+
+	lang, err := client.DetectLanguage(ctx, []string{text})
+	if err != nil {
+		return nil, fmt.Errorf("DetectLanguage: %v", err)
+	}
+	return &lang[0][0], nil
```

Comment:
Is `DetectLanguage` guaranteed to return at least one value, and each value to have at least one value? Don't want to panic.

Reply:
The definition says each slice should hold multiple hypotheses for the language

Reply:
Done